agent-inject/agent/annotations_test.go

```diff
@@ -793,3 +794,75 @@ func Test_runAsSameID(t *testing.T) {
 		})
 	}
 }
+
+func TestInjectContainers(t *testing.T) {
+	expectedPatch := "{\"op\":\"add\",\"path\":\"/spec/containers/0/volumeMounts/-\",\"value\":{\"mountPath\":\"/vault/secrets\",\"name\":\"vault-secrets\"}}"
+
+	tests := []struct {
+		name 		string
+		annotations map[string]string
+		expected    string
+		hasPatch    bool
+	}{
+		{
+			name: 		 "No InjectContainers annotation",
+			annotations: map[string]string{},
+			expected: 	 "foobar",
+			hasPatch:    true,
+		},
+		{
+			name: 		 "InjectContainers annotation with container name",
+			annotations: map[string]string{AnnotationAgentInjectContainers: "baz"},
+			expected: 	 "baz",
+			hasPatch:    false,
+		},
```

Comment:
We'll need another test here for multiple containers being selected via this annotation.